plutus-core/plutus-core/test/TypeSynthesis/Golden/ConstrData.plc.golden

```diff
@@ -1 +1 @@
-(fun (con integer) (fun (con list (data)) (con data)))
\ No newline at end of file
+(fun (con integer) (fun [ (con list) (con data) ] (con data)))
```

Comment:
This diff represents what the PR changes. Now we handle built-in types the same way regardless of whether they're fully monomorphized or not. Previously we had to distinguish between the two cases manually by choosing either `SomeConstant` or `SomeConstantPoly` when defining a builtin and now things are just consistent.

Reply:
s/monomorphized/normalized?

Reply:
> we had to distinguish based on monomorphization happening or not, that's what my comment is about

Well, we actually still have to, just in a different place. Writing a note.